protokt-codegen/src/main/kotlin/com/toasttab/protokt/codegen/annotators/DeserializerAnnotator.kt

```diff
@@ -217,32 +202,59 @@ private constructor(
     )
 
     private fun oneofDes(f: Oneof, ff: StandardField) =
-        OneofTemplate.Deserialize.render(
-            oneof = f.name,
-            name = f.fieldTypeNames.getValue(ff.name),
-            read = deserializeString(ff, ctx, false)
-        )
+        "${f.name}.${f.fieldTypeNames.getValue(ff.fieldName)}(${deserializeString(ff, ctx, false)})"
```

Comment:
Yikes. (I know this came straight from StringTemplate.)

Reply:
Yup, there are a few messy templates in here that should probably be cleaned up and kotlinpoet-ified.